search-sentiments-map/test/trends-test.js

```diff
@@ -0,0 +1,243 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+const chai = require('chai');
+const assert = chai.assert;
+const sinon = require('sinon');
+const trends = require('./../routes/trends').trends;
+const googleTrends = require('google-trends-api');
+const {Datastore} = require('@google-cloud/datastore');
+
+describe('Trends', function() {
+  describe('RetrieveGlobalTrendsForTimeRange', function() {
+    const CURRENT_DATA_TIME_RANGE_12_HOURS_MS = 12 * 60 * 60000;
+    const TIME_RANGE_1_HALF_MS = CURRENT_DATA_TIME_RANGE_12_HOURS_MS * 1.5;
+    const TIME_RANGE_3_MS = CURRENT_DATA_TIME_RANGE_12_HOURS_MS * 3;
+    let mockResult3;
+    let mockResult1half;
+    let mockTrendsEntries;
+    let currentTime;
+
+    beforeEach(() => {
+      currentTime = Date.now();
+      mockTrendsEntries = [{
+        timestamp: currentTime - TIME_RANGE_1_HALF_MS,
+        trendsByCountry: 'trendsByCountry1.5',
+        globalTrends: 'globalTrends1.5',
+      }, {
+        timestamp: currentTime - TIME_RANGE_3_MS,
+        trendsByCountry: 'trendsByCountry3',
+        globalTrends: 'globalTrends3',
+      }];
+
+      mockResult1half = {
+        timestamp: currentTime - TIME_RANGE_1_HALF_MS,
+        globalTrends: 'globalTrends1.5',
+      };
+      mockResult3 = {
+        timestamp: currentTime - TIME_RANGE_3_MS,
+        globalTrends: 'globalTrends3',
+      }
+
+      // Stub calls to the datastore.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        return [mockTrendsEntries];
+      });
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should retrieve the most recent datastore entry beyond RETRIEVE_RESULTS_TIME_MS',
+        async function() {
+      let results = await trends.retrieveGlobalTrendsForTimeRange(0);
+      assert.deepEqual(results, mockResult1half);
+    });
+
+    it('should retrieve the most recent datastore entry given beyond the given time range', 
+        async function() {
+      let results = await trends.retrieveGlobalTrendsForTimeRange(2);
+      assert.deepEqual(results, mockResult3);
+    });
+  });
+
+  describe('UpdateDailyTrends', function() {
+    let mockDailyTrendsData;
```

Comment:
This variable is only used within the beforeEach block. The declaration can be moved in there instead.